lib/workers.js

```diff
@@ -92,10 +92,18 @@ module.exports = {
             load.startJob('python_worker_active', pc.uuid);
             callback(null, pc);
         } else {
-            debug(`getPythonCaller(): adding to waitingCallbacks`);
-            callback.__load_uuid = uuidv4();
-            load.startJob('python_callback_waiting', callback.__load_uuid);
-            waitingCallbacks.push(callback);
+            if (config.workerUseQueue) {
+                debug(`getPythonCaller(): adding to waitingCallbacks`);
+                callback.__load_uuid = uuidv4();
+                load.startJob('python_callback_waiting', callback.__load_uuid);
+                waitingCallbacks.push(callback);
+            } else {
+                debug(`getPythonCaller(): no workers available, waiting to error`);
+                setTimeout(() => {
+                    debug(`getPythonCaller(): no workers available, reporting error`);
+                    callback(new Error('Server is overloaded. Please try again.'));
```

Comment:
whats the logic here behind waiting to throw an error?

Reply:
To slow down submissions. This will only be triggered if we are overloaded, so we want to apply [backpressure](https://medium.com/@jayphelps/backpressure-explained-the-flow-of-data-through-software-2350b3e77ce7).

Reply:
Interesting article, that makes sense